lambdaclass/mirra_backend
# [GH-470] [Dungeon] Add tests for dungeon buffs/upgrades

## Motivation

#632 added buffs but didn't implement tests since the PR was too long. This PR introduces those tests.
Closes https://github.com/lambdaclass/afk_gacha_game/issues/470

## Summary of changes

- Add integration tests for upgrade purchasing and application in battle through SocketTester
- Add Battle test for stat initial modifiers
- Add Battle test for max_level initial modifiers

## How to test it?

run the tests!
`mix test` 

also this messes with some protobuf messages so check in the client that nothing breaks :P (spoiler alert: nothing breaks)
